gcm/gcm.py

```diff
@@ -350,4 +359,8 @@ def json_request(self, registration_ids, data=None, collapse_key=None,
                     backoff *= 2
             else:
                 break
+
+        if unavailable_err is not None:
+            raise unavailable_err
```

Comment:
I think we should rethink the error handling mechanism. But for now we should be consistent with the plaintext_request method which raises an IOError when every retry atempt fails https://github.com/geeknam/python-gcm/blob/master/gcm/gcm.py#L305